src/nupic/utils/CSVHelpers.hpp

```diff
@@ -0,0 +1,139 @@
+#ifndef NTA_UTILS_CSVHELPERS
+#define NTA_UTILS_CSVHELPERS
+
+#include <iostream>
+#include <boost/tokenizer.hpp>
+#include <vector>
+#include <string>
+#include <memory>
+#include <fstream>
+
+#include <nupic/types/Types.hpp>
+//#include "nupic/utils/VectorHelpers.hpp"
+
+namespace nupic {
+namespace utils {
+/** 
+* defines I/O classes for CSV - CSVReader & CSVWriter
+*/
+namespace csv {
+
+//using namespace nupic::utils;
+// read
+template<class T>
+class CSVReader
+{
+    public:
+        CSVReader(std::string filename, UInt skipNHeaderLines=0)
+{
+    this->lineIterator_.reset(new std::ifstream(filename.c_str()));
+    if (!this->lineIterator_->is_open()) {
+        throw "Invalid file";
+    }
+    this->reset_ = this->lineIterator_;
```

Comment:
FIXME: the `reset_` needs to be properly assigned. Used in repeated calls of `readColumn`